dashboard/modules/snapshot/snapshot_head.py

```diff
@@ -131,10 +133,18 @@ async def get_serve_info(self) -> Dict[str, Any]:
                                          gcs_client) or "{}".encode("utf-8")
             deployments_per_controller.append(
                 json.loads(val_bytes.decode("utf-8")))
+        # Merge the deployments dicts of all controllers.
         deployments: Dict[str, Any] = {
             k: v
             for d in deployments_per_controller for k, v in d.items()
         }
+        # Replace the keys (deployment names) with their hashes to prevent
+        # collisions caused by the automatic conversion to camelcase by the
+        # dashboard agent.
+        deployments = {
+            hashlib.sha1(name.encode()).hexdigest(): info
```

Comment:
just a comment here, since we are just hashing a string to avoid duplicate entries, you can technically just use python hash(str)->int, and it should be safe enough.

Reply:
I looked into this and the python hash is different across different runs and different processes, so I don't think we want it here.  (For example, the dashboard agent's hash("my_str") wouldn't match the serve controller's hash("my_str"), and if we restart the controller the hashes would change again.)